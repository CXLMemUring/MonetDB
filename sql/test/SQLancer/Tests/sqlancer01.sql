CREATE TABLE tbl_ProductSales (ColID int, Product_Category  varchar(64), Product_Name  varchar(64), TotalSales int); 
INSERT INTO tbl_ProductSales VALUES (1,'Game','Mobo Game',200),(2,'Game','PKO Game',400),(3,'Fashion','Shirt',500),(4,'Fashion','Shorts',100);
CREATE TABLE another_T (col1 INT, col2 INT, col3 INT, col4 INT, col5 INT, col6 INT, col7 INT, col8 INT);
INSERT INTO another_T VALUES (1,2,3,4,5,6,7,8), (11,22,33,44,55,66,77,88), (111,222,333,444,555,666,777,888), (1111,2222,3333,4444,5555,6666,7777,8888);
CREATE TABLE integers(i INTEGER);
INSERT INTO integers VALUES (1), (2), (3), (NULL);

START TRANSACTION;
-- Bug 6883
select 1 from integers where (((0.7161494983624688) in (integers.i)) = true) = false;
	-- 1
	-- 1
	-- 1

delete from integers where (((0.7161494983624688) in (integers.i)) = true) = false;
	-- 3 rows affected
select i from integers;
	-- NULL

CREATE TABLE t0 (a1 INT, a2 int, a3 int); --Bug 6884
UPDATE t0 SET a2 = (- (- (t0.a1))), a3 = (ascii(CAST(ascii('}悂Y8K*韖5<c>^n8_X1X|p(''bX') AS STRING(920)))) WHERE NOT (((0.27023923567918073) NOT IN (0.9149447665258329)) = FALSE);
	--0 rows affected

insert into integers(i) values(((length(reverse('8 \rcr੧[bp1eMY쫺4j5s뮯!<Rn4*}')))>>(((- (- (528640982)))&(ascii('528640982')))))); --Bug 6885
SELECT i from integers order by i;
	-- NULL
	-- 0
select reverse('8 \rcr੧[bp1eMY쫺4j5s뮯!<Rn4*}');

-- Bug 6886
INSERT INTO another_t(col4) VALUES(-589206676), (-1557408577);
DELETE FROM another_t WHERE ((another_t.col8)<=(+ (another_t.col8)));
ALTER TABLE another_t ADD UNIQUE(col8, col1, col6, col3);
ROLLBACK;

DROP TABLE tbl_ProductSales;
DROP TABLE another_T;
DROP TABLE integers;
