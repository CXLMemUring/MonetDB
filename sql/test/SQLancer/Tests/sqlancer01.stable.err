stderr of test 'sqlancer01` in directory 'sql/test/SQLancer` itself:


# 11:08:10 >  
# 11:08:10 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=30085" "--set" "mapi_usock=/var/tmp/mtest-117725/.s.monetdb.30085" "--forcemito" "--dbpath=/home/ferreira/repositories/MonetDB-Jun2020/BUILD/var/MonetDB/mTests_sql_test_SQLancer" "--set" "embedded_c=true"
# 11:08:10 >  

MAPI  = (monetdb) /var/tmp/mtest-328612/.s.monetdb.34214
QUERY = ALTER TABLE t0 ALTER c0 SET DEFAULT (0.1) IS NULL;
ERROR = !incorrect default value 'SQL feature not yet available for expressions and default values: IS_NULL'
CODE  = 42000

# 11:08:12 >  
# 11:08:12 >  "mclient" "-lsql" "-ftest" "-tnone" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-117725" "--port=30085"
# 11:08:12 >  


# 11:08:12 >  
# 11:08:12 >  "Done."
# 11:08:12 >  

