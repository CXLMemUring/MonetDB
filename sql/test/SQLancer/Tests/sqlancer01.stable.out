stdout of test 'sqlancer01` in directory 'sql/test/SQLancer` itself:


# 11:08:10 >  
# 11:08:10 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=30085" "--set" "mapi_usock=/var/tmp/mtest-117725/.s.monetdb.30085" "--forcemito" "--dbpath=/home/ferreira/repositories/MonetDB-Jun2020/BUILD/var/MonetDB/mTests_sql_test_SQLancer" "--set" "embedded_c=true"
# 11:08:10 >  

# MonetDB 5 server v11.37.0 (hg id: a964bf319dab)
# This is an unreleased version
# Serving database 'mTests_sql_test_SQLancer', using 8 threads
# Compiled for x86_64-pc-linux-gnu/64bit with 128bit integers
# Found 15.496 GiB available main-memory of which we use 12.629 GiB
# Copyright (c) 1993 - July 2008 CWI.
# Copyright (c) August 2008 - 2020 MonetDB B.V., all rights reserved
# Visit https://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://localhost.localdomain:30085/
# Listening for UNIX domain connection requests on mapi:monetdb:///var/tmp/mtest-117725/.s.monetdb.30085
# MonetDB/GIS module loaded
# SQL catalog created, loading sql scripts once
# loading sql script: 09_like.sql
# loading sql script: 10_math.sql
# loading sql script: 12_url.sql
# loading sql script: 13_date.sql
# loading sql script: 14_inet.sql
# loading sql script: 15_querylog.sql
# loading sql script: 16_tracelog.sql
# loading sql script: 17_temporal.sql
# loading sql script: 18_index.sql
# loading sql script: 20_vacuum.sql
# loading sql script: 21_dependency_views.sql
# loading sql script: 22_clients.sql
# loading sql script: 23_skyserver.sql
# loading sql script: 25_debug.sql
# loading sql script: 26_sysmon.sql
# loading sql script: 27_rejects.sql
# loading sql script: 39_analytics.sql
# loading sql script: 39_analytics_hge.sql
# loading sql script: 40_geom.sql
# loading sql script: 40_json.sql
# loading sql script: 40_json_hge.sql
# loading sql script: 41_md5sum.sql
# loading sql script: 45_uuid.sql
# loading sql script: 46_profiler.sql
# loading sql script: 51_sys_schema_extension.sql
# loading sql script: 58_hot_snapshot.sql
# loading sql script: 60_wlcr.sql
# loading sql script: 61_wlcr.sql
# loading sql script: 72_fits.sql
# loading sql script: 74_netcdf.sql
# loading sql script: 75_lidar.sql
# loading sql script: 75_shp.sql
# loading sql script: 75_storagemodel.sql
# loading sql script: 80_statistics.sql
# loading sql script: 80_udf.sql
# loading sql script: 80_udf_hge.sql
# loading sql script: 81_tracer.sql
# loading sql script: 90_generator.sql
# loading sql script: 90_generator_hge.sql
# loading sql script: 99_system.sql
# MonetDB/SQL module loaded

# 11:08:12 >  
# 11:08:12 >  "mclient" "-lsql" "-ftest" "-tnone" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-117725" "--port=30085"
# 11:08:12 >  

#CREATE TABLE tbl_ProductSales (ColID int, Product_Category  varchar(64), Product_Name  varchar(64), TotalSales int); 
#INSERT INTO tbl_ProductSales VALUES (1,'Game','Mobo Game',200),(2,'Game','PKO Game',400),(3,'Fashion','Shirt',500),(4,'Fashion','Shorts',100);
[ 4	]
#CREATE TABLE another_T (col1 INT, col2 INT, col3 INT, col4 INT, col5 INT, col6 INT, col7 INT, col8 INT);
#INSERT INTO another_T VALUES (1,2,3,4,5,6,7,8), (11,22,33,44,55,66,77,88), (111,222,333,444,555,666,777,888), (1111,2222,3333,4444,5555,6666,7777,8888);
[ 4	]
#CREATE TABLE integers(i INTEGER);
#INSERT INTO integers VALUES (1), (2), (3), (NULL);
[ 4	]
#START TRANSACTION;
#select 1 from integers where (((0.7161494983624688) in (integers.i)) = true) = false;
% .%4 # table_name
% %4 # name
% tinyint # type
% 1 # length
[ 1	]
[ 1	]
[ 1	]
#delete from integers where (((0.7161494983624688) in (integers.i)) = true) = false;
[ 3	]
#select i from integers;
% sys.integers # table_name
% i # name
% int # type
% 1 # length
[ NULL	]
#CREATE TABLE t0 (a1 INT, a2 int, a3 int); --Bug 6884
#UPDATE t0 SET a2 = (- (- (t0.a1))), a3 = (ascii(CAST(ascii('}悂Y8K*韖5<c>^n8_X1X|p(''bX') AS STRING(920)))) WHERE NOT (((0.27023923567918073) NOT IN (0.9149447665258329)) = FALSE);
[ 0	]
#insert into integers(i) values(((length(reverse('8 \rcr੧[bp1eMY쫺4j5s뮯!<Rn4*}')))>>(((- (- (528640982)))&(ascii('528640982')))))); --Bug 6885
[ 1	]
#SELECT i from integers order by i;
% sys.integers # table_name
% i # name
% int # type
% 1 # length
[ NULL	]
[ 0	]
#select reverse('8 \rcr੧[bp1eMY쫺4j5s뮯!<Rn4*}');
% .%2 # table_name
% %2 # name
% clob # type
% 28 # length
[ "}*4nR<!뮯s5j4쫺YMe1pb[੧rc\r 8"	]
#INSERT INTO another_t(col4) VALUES(-589206676), (-1557408577);
[ 2	]
#DELETE FROM another_t WHERE ((another_t.col8)<=(+ (another_t.col8)));
[ 4	]
#ALTER TABLE another_t ADD UNIQUE(col8, col1, col6, col3);
#ROLLBACK;
#DROP TABLE tbl_ProductSales;
#DROP TABLE another_T;
#DROP TABLE integers;

# 11:08:12 >  
# 11:08:12 >  "Done."
# 11:08:12 >  

